components/ntp_sponsored_images/browser/ntp_sponsored_images_service.cc

```diff
@@ -165,12 +169,15 @@ void NTPSponsoredImagesService::OnGetPhotoJsonData(
       images_data_->logo_destination_url = *logo_destination_url;
     }
 
-    if (auto* wallpaper_image_urls =
+    if (auto* wallpapers =
             photo_value->FindListPath(kWallpapersPath)) {
-      for (const auto& value : wallpaper_image_urls->GetList()) {
-        images_data_->wallpaper_image_files.push_back(
+      for (const auto& wallpaper : wallpapers->GetList()) {
+        images_data_->backgrounds.push_back({
             base_dir.AppendASCII(
-                *value.FindStringPath(kWallpaperImageURLPath)));
+                *wallpaper.FindStringPath(kWallpaperImageURLPath)),
+            { wallpaper.FindIntPath(kWallpaperFocalPointXPath).value_or(0),
```

Comment:
I think focal point should be the center of the image when it is not present in the JSON. Should we be storing some kind of `null` value instead? @deeppandya is android already setup to use center of image when focal point property is missing?

Reply:
I think client should handle if focal point is not provided. service set zero by default.